packages/source-fetcher/lib/etherscan.ts

```diff
@@ -134,10 +155,9 @@ const EtherscanFetcher: FetcherConstructor = class EtherscanFetcher
       //a testnet for one of the above scanners;
       //part1 is the testnet name, part2 is the broader mainnet name
       let [testnet, network] = [part1, part2];
-      if (network === "arbitrum" && testnet === "rinkeby") {
-        //special case: arbitrum rinkeby is testnet.arbiscan.io,
-        //not rinkeby.arbiscan.io
-        //note: if we supported avalanche, it would have a similar special case
+      if (exceptionalTestnets.has(this.networkName)) {
+        //special case: certain exceptional testnets just use "testnet"
+        //rather than the specific testnet name
```

Comment:
Is it possible to consolidate all this metadata in one data structure? There's a lot of logic to perform what seems to be a simple table lookup to resolve a scanner endpoint. It would simplify adding/removing networks if only a table has to be updated.

Reply:
Eh, OK, I'll go change it to use a lookup table.

Reply:
(The pro of the existing approach is that in many cases you *don't* even have to update a whole table, you can often update less than that. :)  Also, you're less likely to enter something incorrectly when updating it because the things you're entering are smaller and more atomic.  But yes it does raise the specter that special cases could balloon in the future...)

Reply:
OK, I've gone ahead and made this change.